commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommercePriceListAccountGroupTest.java

```diff
@@ -88,26 +95,120 @@ public void testPriceListPriority() throws Exception {
-			_group.getGroupId(), 1, _user.getUserId());
+			_group.getGroupId(), 1, commerceAccountGroup);
 
 		CommercePriceListTestUtil.addUserPriceList(
-			_group.getGroupId(), 2, _user.getUserId());
+			_group.getGroupId(), 2, commerceAccountGroup);
 
 		CommercePriceList expectedCommercePriceList =
 			CommercePriceListTestUtil.addUserPriceList(
-				_group.getGroupId(), 3, _user.getUserId());
+				_group.getGroupId(), 3, commerceAccountGroup);
 
 		Optional<CommercePriceList> actualCommercePriceList =
 			CommercePriceListTestUtil.getCommercePriceList(
-				_group.getGroupId(), 0, _user.getUserId());
+				_group.getGroupId(), _commerceAccount.getCommerceAccountId(),
+				_user.getUserId());
 
 		Assert.assertEquals(
 			expectedCommercePriceList.getCommercePriceListId(),
 			actualCommercePriceList.get(
 			).getCommercePriceListId());
 	}
 
+	@Test(expected = NoSuchElementException.class)
+	public void testPriceListWithDifferentAccountGroups() throws Exception {
+		frutillaRule.scenario(
+			"Add a price list with different account groups"
+		).given(
+			"A group and a user"
+		).and(
+			"I add a price list"
+		).and(
+			"I associate to the price list one account group"
+		).and(
+			"The account trying to get the price list is in a different " +
+				"account group to the one associated to it"
+		).when(
+			"I try to get the price list"
+		).then(
+			"The price list should be returned only if the account belongs " +
+				"to the account group associated to it"
+		);
+
+		CommerceAccountGroupTestUtil.addCommerceAccountToAccountGroup(
+			_commerceAccount);
+
+		CommerceAccountGroup associatedAccountGroup =
+			CommerceAccountGroupTestUtil.addCommerceAccountGroup();
+
+		CommercePriceListTestUtil.addUserPriceList(
+			_group.getGroupId(), 1, associatedAccountGroup);
+
+		Optional<CommercePriceList> actualCommercePriceList =
+			CommercePriceListTestUtil.getCommercePriceList(
+				_group.getGroupId(), _commerceAccount.getCommerceAccountId(),
+				_user.getUserId());
+
+		actualCommercePriceList.get();
+	}
+
+	@Test(expected = Test.None.class)
+	public void testPriceListWithMultipleAccountGroups() throws Exception {
+		frutillaRule.scenario(
+			"Add a price list with multiple account groups"
+		).given(
+			"A group and a user"
+		).and(
+			"I add a price list"
+		).and(
+			"I associate to the price list to many account groups"
+		).when(
+			"I try to get the price list"
+		).then(
+			"The price list should be returned only if the account belongs " +
+				"to every account group associated to it"
+		);
+
+		CommerceAccountGroup accountGroup1 =
+			CommerceAccountGroupTestUtil.addCommerceAccountToAccountGroup(
+				_commerceAccount);
+		CommerceAccountGroup accountGroup2 =
+			CommerceAccountGroupTestUtil.addCommerceAccountToAccountGroup(
+				_commerceAccount);
+		CommerceAccountGroup accountGroup3 =
+			CommerceAccountGroupTestUtil.addCommerceAccountToAccountGroup(
+				_commerceAccount);
+		CommerceAccountGroup accountGroup4 =
```

Comment:
These should all be named `commerceAccountGroupX` rather than just `accountGroupX`.